src/FixedDataTable.js

```diff
@@ -722,49 +736,38 @@ var FixedDataTable = React.createClass({
   },
 
   _onColumnReorderEnd(/*object*/ props, /*object*/ event) {
-    var columnBefore = this.state.columnReorderingData.columnBefore;
-    var columnAfter = this.state.columnReorderingData.columnAfter;
-    var reorderColumn = this.state.columnReorderingData.columnKey;
-    var cancelReorder = this.state.columnReorderingData.cancelReorder;
+    const {
+      columnActions,
+      columnReorderingData: {
+        cancelReorder,
+        columnAfter,
+        columnBefore,
+        columnKey,
+        scrollStart,
+      },
+      onColumnReorderEndCallback,
+      onHorizontalScroll,
+      scrollX,
+    } = this.props;
 
-    this.props.columnActions.stopColumnReorder();
+    columnActions.stopColumnReorder();
     if (cancelReorder) {
-      return
+      return;
     }
 
-    this.props.onColumnReorderEndCallback({
-      columnBefore, columnAfter, reorderColumn
+    onColumnReorderEndCallback({
+      columnAfter,
+      columnBefore,
+      reorderColumn: columnKey,
     });
 
-    var onHorizontalScroll = this.props.onHorizontalScroll;
-    if (this.state.columnReorderingData.scrollStart !== this.state.scrollX && onHorizontalScroll) {
-      onHorizontalScroll(this.state.scrollX)
+    if (scrollStart !== scrollX && onHorizontalScroll) {
+      onHorizontalScroll(scrollX)
     };
   },
 
-  _calculateState(props, oldState) {
```

Comment:
State is dead, wooooo.

Replaced by reselect, but I still need to fix lots of bad merge w/ master issues.  Will do that in a follow up and delete commented out code here after